### ISSUE TITLE
azurerm_mssql_database - support for the restore_long_term_retention_backup_id property is not working

### ISSUE THREAD

**Issue by mglarauzo** [hashicorp/terraform-provider-azurerm#25163]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.6

### AzureRM Provider Version

3.94.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_database

### Terraform Configuration Files

```hcl
resource "azurerm_mssql_database" "example_ltr" {
  name           = "testpitrdb"
  server_id      = "/subscriptions/000000000-0000-0000-0000-000000000000/resourceGroups/resgroup1/providers/Microsoft.Sql/servers/testsqlsrv"
  collation      = "SQL_Latin1_General_CP1_CI_AS"
  max_size_gb    = 2
  sku_name       = "S0"
  create_mode    = "RestoreLongTermRetentionBackup"
  restore_long_term_retention_backup_id = "anystring"
}
```

### Debug Output/Panic Output

```shell
Error: 'long_term_retention_backup_id' is required for create_mode RestoreLongTermRetentionBackup
│
│   with azurerm_mssql_database.example_ltr,
│   on sqldb_restore.tf line 22, in resource "azurerm_mssql_database" "example_ltr":
│   22: resource "azurerm_mssql_database" "example_ltr" {
```

### Expected Behaviour

A new database should be created with data restored from the specified LTR backup.

### Actual Behaviour

Error output asking for an attribute named 'long_term_retention_backup_id'.

Resource docs and schema contain a new attribute named 'restore_long_term_retention_backup_id', but it seems that resource implementation at "internal/services/mssql/mssql_database_resource.go" looks for a different name during validations:

	case databases.CreateModeRestoreLongTermRetentionBackup:
		if _, dbok := d.GetOk("long_term_retention_backup_id"); !dbok {
			return fmt.Errorf("'long_term_retention_backup_id' is required for create_mode %s", createMode)
		}

### Steps to Reproduce

terraform apply

### References

#24904

**Comment by sinbai** [comment]
Hi @mglarauzo thanks for opening this issue. [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/25180) has been submitted to fix this issue. Could you please track it for more updates?

**Comment by mglarauzo** [comment]
Thanks, @sinbai. Already subscribed to the PR!

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #25180. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
